packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -169,6 +175,37 @@ export class QueuedRequestController extends BaseController<
       `${controllerName}:enqueueRequest`,
       this.enqueueRequest.bind(this),
     );
+
+    this.messagingSystem.subscribe(
+      SelectedNetworkControllerEventTypes.stateChange,
+      (_, patch) => {
+        patch.forEach(({ op, path }) => {
+          if (
+            (op === 'replace' || op === 'add') &&
+            path.length === 2 &&
+            path[0] === 'domains' &&
+            typeof path[1] === 'string'
+          ) {
+            this.#flushQueueForOrigin(path[1]);
+          }
```

Comment:
just want to verify if this needs to be op specific (do we need to do anything special for remove?)

Reply:
I guarded this based on the operation because: 
1. I didnt know what to do in the event that the op is 'remove'. Does that imply the network might be changing? Seemed better to not handle it for now.
2. I dont know if the other operations will have the same `path` structure (i.e. will `path[1]` still be the origin? probably... but its better (and faster) to not guess, and remove this code when its more obvious.)